### PR TITLE
Decompile/port new_game

### PR DESCRIPTION
**Pull request by ultima-soul**

### CODE REVIEW

File: src/new_game.c [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    ResetPokeJumpResults();
+    CpuFill16(0, &gSaveBlock2Ptr->berryPick, sizeof(struct BerryPickingResults));
+}
+	
+	
+	
+	
+	
+	
+
```
Comment (Delta456) [inline src/new_game.c]: Why so much space left here?

File: include/union_room_chat.h [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#ifndef GUARD_UNION_ROOM_CHAT_H
+#define GUARD_UNION_ROOM_CHAT_H
+
+void sub_801DD98(void);
+void copy_strings_to_sav1(void);
+
+#endif // GUARD_UNION_ROOM_CHAT_H
```
Comment (Delta456) [inline include/union_room_chat.h]: newline after `#endif`

File: include/battle_records.h [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+extern u8 gRecordsWindowId;
+
+void ClearPlayerLinkBattleRecords(void);
+void InitLinkBattleRecords(void);
+void UpdatePlayerLinkBattleRecords(s32 battlerId);
+void ShowLinkBattleRecords(void);
+void RemoveRecordsWindow(void);
+void ShowTrainerHillRecords(void);
+
+#endif // GUARD_BATTLE_RECORDS_H
```
Comment (Delta456) [inline include/battle_records.h]: newline after `#endif`

File: sym_ewram.txt [hunk 4]
```diff
@@ -485,7 +485,7 @@ gUnknown_2031DA8: @ 2031DA8
 gUnknown_2031DAC: @ 2031DAC
 	.space 0x4
 
-gUnknown_2031DB0: @ 2031DB0
+gDifferentSaveFile: @ 2031DB0
```
Comment (jiangzhengwenjz) [inline sym_ewram.txt]: should be included

File: src/berry_powder.c [hunk 5]
```diff
@@ -66,7 +66,7 @@ bool8 sub_815EF20(u32 a0)
     else
     {
         u32 amount = sub_815EE3C(ptr);
```
Comment (PikalaxALT) [inline src/berry_powder.c]: GetBerryPowder

File: src/new_game.c [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+#include "berry.h"
+#include "easy_chat.h"
+#include "union_room_chat.h"
+#include "mevent.h"
+#include "trainer_tower.h"
+#include "script.h"
+#include "berry_powder.h"
+#include "pokemon_jump.h"
+
+extern const u8 EventScript_ResetAllMapFlags[];
```
Comment (jiangzhengwenjz) [inline src/new_game.c]: This declaration should be in `event_scripts.h`

File: src/new_game.c [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+
+void Sav2_ClearSetDefault(void)
+{
+    ClearSav2();
+    SetDefaultOptions();
+}
+
+void ResetMenuAndMonGlobals(void)
+{
+    gDifferentSaveFile = 0;
```
Comment (jiangzhengwenjz) [inline src/new_game.c]: FALSE

File: src/new_game.c [hunk 8]
```diff
@@ -0,0 +1,163 @@
...
+    sub_8083214(Random());
+    sub_806E6FC();
+}
+
+void NewGameInitData(void)
+{
+    u8 rivalName[PLAYER_NAME_LENGTH];
+
+    StringCopy(rivalName, gSaveBlock1Ptr->rivalName);
+    gDifferentSaveFile = 1;
```
Comment (jiangzhengwenjz) [inline src/new_game.c]: TRUE

File: src/new_game.c [hunk 9]
```diff
@@ -0,0 +1,163 @@
...
+}
+
+static void ResetMiniGamesResults(void)
+{
+    CpuFill16(0, &gSaveBlock2Ptr->berryCrush, sizeof(struct BerryCrush));
+    SetBerryPowder(&gSaveBlock2Ptr->berryCrush.berryPowderAmount, 0);
+    ResetPokeJumpResults();
+    CpuFill16(0, &gSaveBlock2Ptr->berryPick, sizeof(struct BerryPickingResults));
+}
+
```
Comment (jiangzhengwenjz) [inline src/new_game.c]: No newline at end of file

File: src/new_game.c [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+static void ClearPokedexFlags(void)
+{
+    memset(&gSaveBlock2Ptr->pokedex.owned, 0, sizeof(gSaveBlock2Ptr->pokedex.owned));
+    memset(&gSaveBlock2Ptr->pokedex.seen, 0, sizeof(gSaveBlock2Ptr->pokedex.seen));
+}
+
+static void sub_80549D4(void)
+{
+    CpuFill32(0, &gSaveBlock2Ptr->field_B0, (u32) &gSaveBlock2Ptr->mapView - (u32) &gSaveBlock2Ptr->field_B0);
+}
```
Comment (PikalaxALT) [inline src/new_game.c]: The homologous function in Emerald is `ClearFrontierRecord`, so it's highly likely that the 0x7E8 bytes at SaveBlock2+0xB0 is its own struct.

File: include/pokemon_jump.h [hunk 11]
```diff
@@ -0,0 +1,14 @@
+#ifndef GUARD_POKEMON_JUMP_H
+#define GUARD_POKEMON_JUMP_H
+
+#include "main.h"
+
+void sub_802A9A8(u16 monId, MainCallback callback);
+bool32 sub_802C908(u16 species);
+void sub_802C920(void);
+void ResetPokeJumpResults(void);
```
Comment (jiangzhengwenjz) [inline include/pokemon_jump.h]: I think only this declaration makes sense in this file. Those `sub_*` are not even functions in firered.

File: include/new_game.h [hunk 12]
```diff
@@ -1,19 +1,10 @@
 #ifndef GUARD_NEW_GAME_H
 #define GUARD_NEW_GAME_H
 
-#include "global.h"
```
Comment (jiangzhengwenjz) [inline include/new_game.h]: Is there any reason to delete this line?

Comment (ultima-soul) [reply include/new_game.h]: It's in the .c file and it's not present in pokeemerald's new_game.h

File: include/pokemon_jump.h (hunk 11 above)
Comment (ultima-soul) [reply include/pokemon_jump.h]: Yeah, quite a few .h files have pokeemerald function names in them so we should look at cleaning them up eventually.

File: include/new_game.h (hunk 12 above)
Comment (jiangzhengwenjz) [reply include/new_game.h]: It's still better to be there as a dependency.

File: src/new_game.c (hunk 9 above)
Comment (ultima-soul) [reply src/new_game.c]: I thought every file was supposed to have a new line?

File: include/pokemon_jump.h (hunk 11 above)
Comment (jiangzhengwenjz) [reply include/pokemon_jump.h]: Yes, most should be lazy copies at an early stage. but if we know they're incorrect, there's no reason to include them.

File: src/new_game.c (hunk 9 above)
Comment (jiangzhengwenjz) [reply src/new_game.c]: Every text file should end with newline. i.e. Line 163 should be empty.

**Pull request merged**